auth_proxy/services.py

```diff
@@ -53,6 +54,25 @@ def audit(self, client_id):
         return self.db.session.query(Token).\
             filter_by(client_id=client_id)
 
+    def smart_token_credentials(self, grant_type, code=None, refresh_token=None):
+        """ Provide additional credentials required by a SMART token request.
+        """
+        if grant_type == 'authorization_code':
+            grant = self.db.session.query(Grant).\
+                filter_by(code=code).first()
+            user = grant.user
```

Comment:
Maybe I missed something, this is just making sure that the patient id is present in the response to `/token`. There isn't a request.user to compare against. I think the place we actually want that logic is line 93 of this file, where I'm querying for `id=1`.